src/Adapter/Product/Combination/Update/CombinationStockUpdater.php

```diff
@@ -138,54 +128,46 @@ private function fillUpdatableProperties(Combination $combination, CombinationSt
      */
     private function updateStockAvailable(Combination $combination, CombinationStockProperties $properties): void
     {
-        $updateQuantity = null !== $properties->getQuantity();
+        $updateQuantity = null !== $properties->getDeltaQuantity();
         $updateLocation = null !== $properties->getLocation();
 
         if (!$updateQuantity && !$updateLocation) {
             return;
         }
 
-        $newQuantity = $properties->getQuantity();
-        $newLocation = $properties->getLocation();
-
         $stockAvailable = $this->stockAvailableRepository->getForCombination(new CombinationId((int) $combination->id));
 
         if ($updateQuantity) {
-            $this->saveMovement($combination, (int) $stockAvailable->quantity, $newQuantity);
-            $stockAvailable->quantity = $newQuantity;
+            $this->updateQuantity($stockAvailable, $properties->getDeltaQuantity());
         }
 
         if ($updateLocation) {
-            $stockAvailable->location = $newLocation;
+            $stockAvailable->location = $properties->getLocation();
         }
 
         $this->stockAvailableRepository->update($stockAvailable);
     }
 
     /**
-     * @param Combination $combination
-     * @param int $oldQuantity
-     * @param int $newQuantity
-     *
-     * @throws CoreException
+     * @param StockAvailable $stockAvailable
+     * @param DeltaQuantity $deltaQuantity
      */
-    private function saveMovement(Combination $combination, int $oldQuantity, int $newQuantity): void
+    private function updateQuantity(StockAvailable $stockAvailable, DeltaQuantity $deltaQuantity): void
     {
-        $combinationId = $combination->id;
-        $deltaQuantity = $newQuantity - $oldQuantity;
+        $deltaQuantityValue = $deltaQuantity->getDeltaQuantity();
+        $stockAvailable->quantity += $deltaQuantityValue;
 
-        if (0 === $deltaQuantity) {
+        if (0 === $deltaQuantityValue) {
             return;
         }
 
-        try {
-            $this->stockManager->saveMovement($combination->id_product, $combinationId, $deltaQuantity);
-        } catch (PrestaShopException $e) {
-            throw new CoreException(
-                sprintf('Error occurred when trying to save stock movement for combination %d', $combinationId),
-                0,
-                $e
-            );
-        }
+        $this->stockManager->saveMovement(
```

Comment:
This part should be moved into the `updateStockAvailable` after the `$stockAvailable` has been successfully updated, and only when `$updateQuantity` is true of course